src/components/error/idle-warning.tsx

```diff
@@ -0,0 +1,50 @@
+import React, { useEffect, useState } from "react";
+import "./error.scss";
+
+interface IProps {
+  username: string;
+  onTimeout: () => void;
+  onContinue: () => void;
+  onExit: () => void;
+  timeout: number;
+}
+
+export const IdleWarning: React.FC<IProps> = ({ username, timeout, onTimeout, onContinue, onExit }) => {
+  const [ time, setTime ] = useState(0);
+
+  useEffect(() => {
+    const startTime = Date.now();
+    const intervalId = setInterval(() => {
+      const currentTime = Date.now();
+      const timeDiff = currentTime - startTime;
+      setTime(timeDiff);
+      if (timeDiff >= timeout) {
+        window.clearInterval(intervalId);
+      }
+    }, Math.min(timeout / 2, 500));
```

Comment:
Should this 500 be stored in a constant?

Reply:
Thanks, fixed.